third_party/amd/lib/TritonAMDGPUToLLVM/DotOpToLLVM/WMMA.cpp

```diff
@@ -227,39 +226,60 @@ std::string addInstructionSuffix(std::string intrinsicName, unsigned kWidth,
 }
 
 Value generateWMMAIntrinsic(ConversionPatternRewriter &rewriter, Location loc,
-                            Value valA, Value valB, Value valC, Type aElType,
-                            Type bElType, Type dElType, StringRef name,
-                            std::optional<bool> tiedLower) {
+                            int wmmaVer, Value valA, Value valB, Value valC,
+                            Type aElType, Type bElType, Type dElType,
+                            StringRef name, std::optional<bool> tiedLower) {
   auto b = TritonLLVMOpBuilder(loc, rewriter);
 
   LLVM::FastmathFlagsAttr defaultFlags{};
   SmallVector<Value> operands;
+  int64_t kWidth = cast<VectorType>(valA.getType()).getNumElements();
+
   if (aElType.isInteger())
     operands.push_back(b.int_val(1, !aElType.isUnsignedInteger()));
+  if (wmmaVer == 3 && (kWidth == 16 && (aElType.isBF16() || aElType.isF16())))
+    operands.push_back(b.int_val(1, 0));
   operands.push_back(valA);
+
   if (bElType.isInteger())
     operands.push_back(b.int_val(1, !bElType.isUnsignedInteger()));
+  if (wmmaVer == 3 && (kWidth == 16 && (bElType.isBF16() || bElType.isF16())))
+    operands.push_back(b.int_val(1, 0));
   operands.push_back(valB);
+
+  if (wmmaVer == 3 &&
+      (kWidth == 16 || (kWidth == 8 && aElType.getIntOrFloatBitWidth() == 8)))
+    operands.push_back(b.int_val(16, 0));
+
   operands.push_back(valC);
+
   // Flag for using low bits in registers. Result could be already packed to
   // int32. Set low bits by default for now.
   if (tiedLower.has_value() || 32 / dElType.getIntOrFloatBitWidth() > 1 ||
       dElType.isInteger(32)) {
     operands.push_back(b.int_val(1, tiedLower.value_or(false)));
   }
+
+  // add two additional operands perf llvm changes
```

Comment:
typo: per

Reply:
Yeah would be better to make it clear with comments like "for LLVM gfx1250 A/B reuse control"